### PR TITLE
Simplify timers and query forcing in `analysis` query

### PR DESCRIPTION
**Pull request by cjgillot**
`sess.time` is not useful around queries, as they are timed by the query system itself.

Other uses of `sess.time` are replaced by `sess.timer`, to reduce callback pollution and simplify code.

cc @SparrowLii as this changes the parallelization pattern.

### CODE REVIEW

File: compiler/rustc_interface/src/passes.rs [hunk 1]
```diff
@@ -746,81 +746,69 @@ fn analysis(tcx: TyCtxt<'_>, (): ()) -> Result<()> {
...
+            tcx.ensure().stability_index(());
+        }
+    );
 
     // passes are timed inside typeck
     rustc_hir_analysis::check_crate(tcx)?;
 
-    sess.time("MIR_borrow_checking", || {
-        tcx.hir().par_body_owners(|def_id| tcx.ensure().mir_borrowck(def_id));
-    });
```
Comment (bjorn3) [inline compiler/rustc_interface/src/passes.rs:792]: This removes the aggregate count of borrow checking time from `-Ztime-passes`, right? AFAIK queries are not shown by `-Ztime-passes`.

Comment (cjgillot) [reply compiler/rustc_interface/src/passes.rs:792]: Yes. That would remove it.
I wonder if we should remove `-Ztime-passes` entirely. It's already incorrect (some borrow-check may already have happened earlier), and bound to be wronger with parallel compilation.

Comment (bjorn3) [reply compiler/rustc_interface/src/passes.rs:792]: `-Ztime-passes` is much more convenient when trying to find roughly where the bottleneck is. I don't see myself asking someone to install measureme to get a rough idea why rustc is so slow on their code. Also `-Zself-profile` only works if rustc terminates in the first place. If it gets stuck somewhere `-Zself-profile` is the only thing that can provide a guess where it is stuck.

**Comment by SparrowLii** [comment]
Thanks for pushing this!

**Comment by b-naber** [comment]
I think the concern about `-Ztime-passes` is valid, so I don't feel comfortable r+'ing this. Maybe r? @oli-obk ?

File: compiler/rustc_interface/src/passes.rs (hunk 1 above)
Comment (oli-obk) [reply compiler/rustc_interface/src/passes.rs:792]: what we could do here is

* use `tracing-indicatif` for showing progress on queries as the compiler runs
* use a custom tracing subscriber that accumulates data for every query and how long it was run and emits that information at the end.

But I think that's orthogonal to this PR. We've already reduced the usefulness of `-Ztime-passes` for "phase tracking" in previous PRs that did these kind of changes and as @cjgillot said, we've already made it less useful years ago just by introducing queries.

**Comment by oli-obk** [comment]
r=me with timers left in until we replace them with something accurate

**Pull request closed without merging**